JM/CF/Scripts/3_Game/CommunityFramework/Stream/CF_BinaryStream.c

```diff
@@ -0,0 +1,286 @@
+					fs.Write(ivalue);
+				}
+				
+				fs.Close();
+
+				break;
+			}
+		}
+		return true;
+	}
+	
+	void UpdateNonZeroFields()
+	{
+		if (!m_NonZeroesDirty) return;
+		
+		m_NonZeroes.Clear();
+		
+		int needed = m_Data.Count() % 4;
+		int lastZero = 0;
+		
+		for (int index = 0; index < m_Data.Count(); index++)
+		{
+			if (m_NonZeroes.Count() >= needed) break;
+
+			if (m_Data[index] == 0) continue;
+
+			// The value is at the start of a block of 4 bytes
+			if (((index - lastZero) % 4) == 0)
+			{
+				m_NonZeroes.Insert(index);
+				lastZero = index + 1;
+			}
+		}
+
+		m_NonZeroesDirty = false;
+	}
+
+	override void Write(int value)
+	{
+		m_Data.InsertAt(value, m_Position++);
+		
+		m_NonZeroesDirty = true;
+	}
+
+	override void WriteChar(string value)
+	{
+		Write(value.Hash());
+	}
+
+	override void WriteBool(bool value)
+	{
+		if (value) Write(0x00000001);
+		else Write(0x00000000);
+	}
+
+	override void WriteInt(int value)
+	{
+		Write((value >> 24) & 0x000000FF);
+		Write((value >> 16) & 0x000000FF);
+		Write((value >> 8 ) & 0x000000FF);
+		Write((value	  ) & 0x000000FF);
+	}
+
+	override void WriteFloat(float value)
+	{
+		WriteInt(CF_Cast<float, int>.Reinterpret(value));
+	}
+
+	override void WriteVector(vector value)
+	{
+		WriteFloat(value[0]);
+		WriteFloat(value[1]);
+		WriteFloat(value[2]);
+	}
+
+	override void WriteString(string value)
+	{
+		WriteInt(value.Length());
+		for (int i = 0; i < value.Length(); i++)
+		{
+			Write(value[i].Hash());
+		}
+	}
+
+	override void WriteCString(string value)
+	{
+		for (int i = 0; i < value.Length(); i++)
+		{
+			Write(value[i].Hash());
+		}
+		Write(0);
+	}
+
+	override int Read()
+	{
+		return m_Data[m_Position++];
+	}
+
+	override string ReadChar()
+	{
+		int b = m_Data[m_Position++];
+		return b.AsciiToString();
+	}
+
+	override bool ReadBool()
+	{
+		return Read() != 0;
+	}
+
+	override int ReadInt()
+	{
+		int b3 = Read();
+		int b2 = Read();
+		int b1 = Read();
+		int b0 = Read();
+
+		return ((b3 & 0x000000FF) << 24) + ((b2 & 0x000000FF) << 16) + ((b1 & 0x000000FF) << 8) + (b0);
+	}
+
+	override float ReadFloat()
+	{
+		return CF_Cast<int, float>.Reinterpret(ReadInt());
+	}
+
+	override vector ReadVector()
+	{
+		vector v;
+		v[0] = ReadFloat();
+		v[1] = ReadFloat();
+		v[2] = ReadFloat();
+		return v;
+	}
+
+	override string ReadString()
+	{
+		string str;
+
+		for (int index = 0; index < ReadInt(); index++)
+		{
+			str += Read().AsciiToString();
+		}
+
+		return str;
+	}
+
+	override string ReadCString()
+	{
+		string str;
+
+		int c = Read();
+		while (c != 0)
+		{
+			str += c.AsciiToString();
+			c = Read();
+		}
+
+		return str;
+	}
+
+	override bool EOF()
+	{
+		return m_Position >= m_Data.Count();
+	}
+
+	override int Position()
+	{
+		return m_Position;
+	}
+
+	override int Length()
+	{
+		return m_Data.Count();
+	}
+
+	override void GoTo(int position)
+	{
+		m_Position = position;
+	}
+
+	override void Seek(int offset)
```

Comment:
Prototype should be `void Seek(int offset, CF_Stream_Origin origin)` where origin is an Enum with for example `SEEK_SET`, `SEEK_CUR`, `SEEK_END` ... inspired by [c++ fseek](https://www.cplusplus.com/reference/cstdio/fseek/)

Default value for origin should be `SEEK_CUR`